src/main/java/build/buildfarm/server/BuildFarmServer.java

```diff
@@ -65,6 +69,31 @@ public BuildFarmServer(String session, BuildFarmServerConfig config)
     this(session, ServerBuilder.forPort(config.getPort()), config);
   }
 
+  private Collection<String> getAuthTokens(BuildFarmServerConfig config) throws ConfigurationException {
+    Collection<String> tokens = new ArrayList<>();
+
+    if (config.getAuth().hasStatic()) {
+      tokens.addAll(config.getAuth().getStatic().getTokensList());
+
+      String tokensFile = config.getAuth().getStatic().getTokensFile();
+      if (tokensFile != null && tokensFile != "") {
+        try {
+          List<String> lines = Files.readAllLines(Path.of(tokensFile));
+          for (String line : lines) {
+            String trimmed = line.trim();
+            if (trimmed.startsWith("#") || trimmed.isEmpty())
+              continue;
+            tokens.add(line);
+          }
+        } catch (IOException ex) {
+          throw new ConfigurationException("Unable to read from tokens file: " + ex.getMessage());
```

Comment:
what is the point of catching exception A and throwing exception B?
why not just declare that this method raises an IOException since if this happens we don't want to recover and just crash the process ?

Reply:
Because of Java's "checked execeptions" feature. `Files.readAllLines` declares IOException as a checked exception which requires handling `IOException` or declaring in `getAuthTokens` that it is a permitted checked exception (and it would then have to be handled in the `BuildFarmServer` constructor.) Note that the `BuildFarmServer` constructor declares `InterruptedException` and `ConfigurationException` as checked exceptions. Thus I conformed `getAuthTokens` to match the API alredy preswnted by `BuildFarmServer` to the rest of the source code and confined the IOException to this file.

Reply:
Will the BF server crash/refuse to load if getAuthTokens raises ConfigurationException ?
if this is the case I don't care as much.